### ISSUE TITLE
bug: Unable to install home-manager (NixOS, standalone) "docbook-xsl/default.nix:1:1 called with unexpected argument 'withManOptDedupPatch'"

### ISSUE THREAD

**Issue by sirwobin** [nix-community/home-manager#3148]
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

1. Fresh installation of NixOS under WSL
2. As root added nixpkgs unstable channel with home-manager master channel.
3. As root added `<home-manager/nixos>` to imports in configuration.nix
4. As root executed `nixos-rebuild switch --upgrade`
5. As user executed `nix-shell '<home-manager>' -A install`

Received the following error:
```
nixos@nixos:~/ > nix-shell '<home-manager>' -A install

Creating initial Home Manager generation...

error: anonymous function at /nix/store/nn8bph8rjhh3l0l36mgsvr7zwniapc2k-nixos-21.11.337975.eabc3821918/nixos/pkgs/data/sgml+xml/stylesheets/xslt/docbook-xsl/default.nix:1:1 called with unexpected argument 'withManOptDedupPatch'

       at /nix/store/nn8bph8rjhh3l0l36mgsvr7zwniapc2k-nixos-21.11.337975.eabc3821918/nixos/lib/customisation.nix:133:63:

          132|       pkgs = f origArgs;
          133|       mkAttrOverridable = name: _: makeOverridable (newArgs: (f newArgs).${name}) origArgs;
             |                                                               ^
          134|     in
(use '--show-trace' to show detailed location information)
Uh oh, the installation failed! Please create an issue at

    https://github.com/nix-community/home-manager/issues

if the error seems to be the fault of Home Manager.```

### Maintainer CC

(unable to identify the home-manager module causing the bug)

### System information

```markdown
nixos@nixos:~/ > nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.60.1-microsoft-standard-WSL2, NixOS, 22.11 (Raccoon)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.10.3`
 - channels(root): `"nixos-21.11.337975.eabc3821918, home-manager, nixpkgs-22.11pre400474.441dc5d5121"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by berbiche** [comment]
Hi,

This issue this was resolved months ago and is available in nixpkgs 22.05 and 22.11.
Judging by your channels, your system still has an old 21.11 channel laying around.

I recommend removing this channel with `nix-channel -e nixos`

**Comment by seanybaggins** [comment]
I had to update my nixos channel for both my user and the root user.

```
nix-channel --remove nixos
nix-channel --add https://nixos.org/channels/nixos-unstable nixos
sudo nix-channel --remove nixos
sudo nix-channel --add https://nixos.org/channels/nixos-unstable nixos
nix-channel --update
sudo nix-channel --update
sudo nixos-rebuild switch --upgrade
```
